dynawo/sources/Common/DYNTrace.cpp

```diff
@@ -147,22 +147,22 @@ void Trace::addAppenders_(const std::vector<TraceAppender>& appenders) {
   // remove old appenders (console_log)
   Trace::resetCustomAppenders();
 
+  traceAppenders_.clear();  // clear previous appenders for a new job
   for (const TraceAppender& appender : appenders) {
     traceAppenders_.insert(std::make_pair(appender.getTag(), appender));
   }
 
-  configureSink(appenders);
+  configureSink(appenders, true);
   logging::add_common_attributes();
 }
 
-void Trace::configureSink(const std::vector<TraceAppender>& appenders) {
+void Trace::configureSink(const std::vector<TraceAppender>& appenders, bool eraseSinks) {
   logging::attributes::current_thread_id::value_type currentId =
     logging::attributes::current_thread_id().get_value().extract<logging::attributes::current_thread_id::value_type>().get();
 
-  TraceSinks traceSink;
-  if (sinks_.find(currentId) != sinks_.end()) {
-    TraceSinks& traceSinkOld = sinks_.at(currentId);
-    traceSink.persistantSinks = traceSinkOld.persistantSinks;
+  TraceSinks& traceSink = sinks_.at(currentId);
```

Comment:
why removing the protection for non-existing currentId?

Reply:
I changed the line to this now (because there was a crash because of the at()) :
TraceSinks& traceSink = sinks_[currentId];

I think it's better to modify sinks_ directly, hence the changes I made.